add MacOS 10 support to PyPI package

BitBaseBit opened the pull request.

murcake reviewed (approved):
I didn't know that binary format has changed between 10.15 and 11

BitBaseBit commented:
> I didn't know that binary format has changed between 10.15 and 11

We aren't doing universal builds, I've got two macs , one with 10.15 and one with 11, when I build kudu on 11 it doesn't work on 10.15. Also we are using a bunch of C++ features that are only supported on MacOS 10.15+:
```
 90%] Building CXX object CMakeFiles/warp-solidity.dir/yul_prepass/YulCleaner.cpp.o
/Users/henrilieutaud/Documents/Workshop/Cairo/kudu/solidity_prepass/WarpVisitor.cpp:47:24: error: 'remove' is unavailable: introduced in macOS 10.15
        if (std::filesystem::remove(filePath))
                   ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/filesystem:1864:39: note: 'remove' has been explicitly marked unavailable here
inline _LIBCPP_INLINE_VISIBILITY bool remove(const path& __p) {
                   ^
/Users/henrilieutaud/Documents/Workshop/Cairo/kudu/solidity_prepass/WarpVisitor.cpp:47:31: error: 'path' is unavailable: introduced in macOS 10.15
        if (std::filesystem::remove(filePath))
                      ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/filesystem:773:3: note: 'path' has been explicitly marked unavailable here
 path(const _Source& __src, format = format::auto_format) {
 ^
/Users/henrilieutaud/Documents/Workshop/Cairo/kudu/solidity_prepass/WarpVisitor.cpp:47:31: error: '~path' is unavailable: introduced in macOS 10.15
        if (std::filesystem::remove(filePath))
                      ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/filesystem:791:3: note: '~path' has been explicitly marked unavailable here
 ~path() = default;
 ^
/Users/henrilieutaud/Documents/Workshop/Cairo/kudu/solidity_prepass/WarpVisitor.cpp:52:32: error: 'filesystem_error' is unavailable: introduced in macOS
   10.15
    catch (const std::filesystem::filesystem_error& err)
                   ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/filesystem:1320:61: note: 'filesystem_error' has been explicitly marked unavailable here
class _LIBCPP_AVAILABILITY_FILESYSTEM _LIBCPP_EXCEPTION_ABI filesystem_error : public system_error {
                              ^
```
